src/csharp/Tests/Record.UnitTests/LoopbackTests.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using Microsoft.Azure.Kinect.Sensor;
+using Microsoft.Azure.Kinect.Sensor.Record;
+using NUnit.Framework;
+
+namespace Tests
+{
+    /// <summary>
+    /// Loopback Tests write to a recording, and then read the recording back to verify the API.
+    /// </summary>
+    public class LoopbackTests
+    {
+        private string recordingPath;
+
+        /// <summary>
+        /// Allocate a path for the recording.
+        /// </summary>
+        [SetUp]
+        public void Setup()
+        {
+            this.recordingPath = System.IO.Path.Combine(System.IO.Path.GetTempPath(), "testfile.mkv");
+        }
+
+        /// <summary>
+        /// Delete the temporary recording.
+        /// </summary>
+        [TearDown]
+        public void TearDown()
+        {
+            System.IO.File.Delete(this.recordingPath);
+        }
+
+        /// <summary>
+        /// Writes each of the data types to a file and reads them back.
+        /// Verfies as many properties as possible.
+        /// </summary>
+        [Test]
+        public void LoopbackTest1()
+        {
+            DeviceConfiguration deviceConfiguration = new DeviceConfiguration()
+            {
+                CameraFPS = FPS.FPS30,
+                ColorFormat = ImageFormat.ColorNV12,
+                ColorResolution = ColorResolution.R720p,
+                DepthDelayOffColor = TimeSpan.FromMilliseconds(123),
+                DepthMode = DepthMode.NFOV_2x2Binned,
+                DisableStreamingIndicator = true,
+                SuboridinateDelayOffMaster = TimeSpan.FromMilliseconds(456),
+                SynchronizedImagesOnly = true,
+                WiredSyncMode = WiredSyncMode.Subordinate,
+            };
+
+            using (Recorder record = Recorder.Create(this.recordingPath, null, deviceConfiguration))
+#pragma warning restore CA1508 // Avoid dead conditional code
+            {
+                record.AddImuTrack();
+                record.AddCustomVideoTrack("CUSTOM_VIDEO", "V_CUSTOM1", new byte[] { 1, 2, 3 }, new RecordVideoSettings() { FrameRate = 1, Height = 10, Width = 20 });
+                record.AddCustomSubtitleTrack("CUSTOM_SUBTITLE", "S_CUSTOM1", new byte[] { 4, 5, 6, 7 }, new RecordSubtitleSettings() { HighFrequencyData = false });
+                record.AddTag("MYTAG1", "one");
+                record.AddTag("MYTAG2", "two");
+
+                record.WriteHeader();
```

Comment:
You have a comment that WriteHeader must be called before any data is written, but does it need to be called after all tracks and tags are added?